contracts/CompoundGovernor.sol

```diff
@@ -64,6 +114,69 @@ contract CompoundGovernor is
         __GovernorPreventLateQuorum_init(_initialVoteExtension);
         __GovernorSettableFixedQuorum_init(_quorumVotes);
         __Ownable_init(_initialOwner);
+        _setWhitelistGuardian(_whitelistGuardian);
+        _setProposalGuardian(_proposalGuardian);
+    }
+
+    /// @notice Sets or updates the whitelist expiration for a specific account.
+    /// @param _account The address of the account to be whitelisted.
+    /// @param _expiration The timestamp until which the account will be whitelisted.
+    /// @dev Only the executor (timelock) or the whitelistGuardian can call this function.
+    function setWhitelistAccountExpiration(address _account, uint256 _expiration) external {
+        if (msg.sender != _executor() && msg.sender != whitelistGuardian) {
+            revert Unauthorized("Not timelock or guardian", msg.sender);
+        }
+
+        whitelistAccountExpirations[_account] = _expiration;
+        emit WhitelistAccountExpirationSet(_account, _expiration);
+    }
+
+    /// @notice Checks if an account is currently whitelisted.
+    /// @param _account The address of the account to check.
+    /// @return bool Returns true if the account is whitelisted (expiration is in the future), false otherwise.
+    function isWhitelisted(address _account) public view returns (bool) {
```

Comment:
also wondering if it might be more explicit to call this a proposer whitelist, implying a change of name to `isWhitelistedProposer`

Reply:
We agreed to keep the old naming to reduce the scope of changes made to the contract.